compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -95,7 +94,9 @@ impl<'a, 'hir> ItemLowerer<'a, 'hir> {
         for (def_id, info) in lctx.children {
             self.owners.ensure_contains_elem(def_id, || hir::MaybeOwner::Phantom);
             debug_assert!(matches!(self.owners[def_id], hir::MaybeOwner::Phantom));
-            self.owners[def_id] = info;
+            if let hir::MaybeOwner::NonOwner(_) = self.owners[def_id] {
+                self.owners[def_id] = info;
+            }
```

Comment:
The logic is reversed here.
`Phantom` means nothing has been assigned yet.
An `Owner` can overwrite a `NonOwner`, and a `NonOwner` does not overwrite to an `Owner`.
If we try to overwrite an `Owner` with an `Owner`, or a `NonOwner` with a `NonOwner`, that's a bug.

Reply:
Is it better now?